### PR TITLE
Rename existing pre_commit to pre-commit

### PR DESCRIPTION
**Pull request by nicoddemus**
Rename existing pre_commit to pre-commit

Copied almost verbatim from conda-forge/pre_commit, only changing:

* package name
* build number
* pip install command

See conda-forge/pre_commit-feedstock#35

### CODE REVIEW

File: recipes/pre-commit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - aspy.yaml
+    - cached-property
+    - cfgv >=1.0.0
+    - identify >=1.0.0
+    - importlib_metadata
+    - importlib_resources  # [py<37]
+    - nodeenv >=0.11.1
+    - python
+    - pyyaml
+    - setuptools
```
Comment (marcelotrevisani) [inline recipes/pre-commit/meta.yaml]: Do you really need ``setuptools`` to run this project?

File: recipes/pre-commit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  entry_points:
+    - pre-commit = pre_commit.main:main
+    - pre-commit-validate-config = pre_commit.clientlib:validate_config_main
+    - pre-commit-validate-manifest = pre_commit.clientlib:validate_manifest_main
+
+requirements:
+  build:
```
Comment (marcelotrevisani) [inline recipes/pre-commit/meta.yaml]: Could you please change it to host?
```suggestion
  host:
```

File: recipes/pre-commit/meta.yaml (hunk 1 above)
Comment (nicoddemus) [reply recipes/pre-commit/meta.yaml]: IIRC yes, but let's see. 👍

Comment (nicoddemus) [reply recipes/pre-commit/meta.yaml]: You are right, not needed removed.

File: recipes/pre-commit/meta.yaml (hunk 2 above)
Comment (nicoddemus) [reply recipes/pre-commit/meta.yaml]: Done!

File: recipes/pre-commit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    - cfgv >=1.0.0
+    - identify >=1.0.0
+    - importlib_metadata
+    - importlib_resources  # [py<37]
+    - nodeenv >=0.11.1
+    - python
+    - pyyaml
+    - six
+    - toml
+    - virtualenv
```
Comment (marcelotrevisani) [inline recipes/pre-commit/meta.yaml]: I think you also need ``futures`` for Python 2, right? And ``virtualenv >=15.2`` ?
```suggestion
    - virtualenv >=15.2
    - futures  # [py2k]
```

Comment (nicoddemus) [reply recipes/pre-commit/meta.yaml]: Yep, thanks, good catch!

**Pull request merged**